nix/egil-scim-client.nix

```diff
@@ -0,0 +1,100 @@
+{ version
+, homepage
+, downloadPage
+, changelog
+, maintainers
+, platforms
+, isDebugBuild ? false
+, doCheck ? true
+}:
+{ lib
+, boost
+, cmake
+, curl
+, filterSource
+, openldap
+, stdenv
+, ...
+}:
+
+let
+  inherit (lib)
+    optionalString
+    optionals
+  ;
+
+  packageName = "egil-scim-client";
+  exeName = "EgilSCIMClient";
+
+  suffix = optionalString isDebugBuild "-debug";
+  pname = "${packageName}${suffix}";
+  exePath = "/bin/${exeName}${suffix}";
+in
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = filterSource {
+    src = ./..;
+    filesToKeep = [ "CMakeLists.txt" ];
+    directoriesToKeep = [ "/src" ];
+    name = packageName;
+  };
+
+  outputs = [ "bin" "dev" ] ++ optionals isDebugBuild [ "source" ] ++ [ "out" ];
```

Comment:
```suggestion
  outputs = [ "out" "bin" "dev" ] ++ optionals isDebugBuild [ "source" ];
```

Reply:
That "out" output is actually empty. I found out that it is actually possible to completely remove it by setting a couple of environment variables (like `outputLib` that could be used to specify the which output the libraries should go to), but I didn't like that approach (what should be the value of `outputLib` since there isn't any library produced. Maybe "/dev/null"?).  The main part of this derivation is the binary so that's why I put it as the first output. Surprisingly, this approach doesn't seem to be that common in nixpkgs; most of the time `out` is used rather than semantically correct output but I don't know why.

Reply:
Because of [this indirect redirection](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/multiple-outputs.sh#L86), it is not possible to set, for example, `outputLib` to a value like "/dev/null". So currently, leaving the `out` output seems to be the best option. But I think it shouldn't be the first output since doing that makes it the default output; which is both semantically and functionally wrong. Semantically wrong because the executable is the main component and functionally wrong because the `out` output is empty.